projects/silo/index.js

```diff
@@ -95,16 +162,51 @@ async function getSilos(api) {
     logs = [...logs, ...logChunk];
   }
 
-  return logs.map((log) => `0x${log.topics[1].substring(26)}`).filter((address) => fallbackBlacklist.indexOf(address.toLowerCase()) === -1);
+  return logs.map((log) => `0x${log.topics[1].substring(26)}`).filter((address) => blacklistedSilos.indexOf(address.toLowerCase()) === -1);
+}
+
+async function getSilosV2(api) {
+  const decoder = ethers.AbiCoder.defaultAbiCoder();
+  const chain = api.chain;
+  let logs = [];
+  let siloAddresses = [];
+  if(configV2[chain]) {
+    for(let factory of configV2[chain].factories) {
+      const { SILO_FACTORY, START_BLOCK } = factory;
+      let logChunk = await getLogs({
+        api,
+        target: SILO_FACTORY,
+        fromBlock: START_BLOCK,
+        topic: 'NewSilo(address,address,address,address,address,address)',
```

Comment:
you can skip the decoder and pass the `eventAbi` string here, we automatically decode it.

Reply:
I have adjusted this as suggested, thank you sir